src/pudl/output/pudltabl.py

```diff
@@ -276,8 +274,9 @@ def _select_between_dates(self, table: str) -> sa.sql.expression.Select:
             ``report_date`` or ``report_year``) to lie between ``self.start_date`` and
             ``self.end_date`` (inclusive).
         """
-        pt = pudl.output.pudltabl.get_table_meta(self.pudl_engine)
-        tbl = pt[f"{table}"]
+        md = sa.MetaData()
+        md.reflect(self.pudl_engine)
+        tbl = md.tables[f"{table}"]
```

Comment:
Should we catch a `KeyError` if the table exists? I don't think this check happens prior to this method being called.

Reply:
Do you mean if the table doesn't exist?  Is there a useful action we can take if we get a bad table name, or is it just going to error out anyway? This is only used internal to the PudlTabl class and we're going to nuke it soon and I think it should probably just fail if we have a bad table name.

Reply:
I figured we could catch the `KeyError` and raise a helpful message like `f"{table} not found in the metadata."`

Reply:
Added try-except here